error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -98,6 +99,102 @@ Mono<T> after() {
     }
   }
 
+  /**
+   * Prefer {@link Mono#timeout(Duration, Mono)} over more contrived or less performant
+   * alternatives.
+   */
+  static final class MonoTimeoutMonoEmpty<T> {
+    @BeforeTemplate
+    Mono<T> before(Mono<T> mono, Duration duration) {
+      return mono.timeout(duration).onErrorComplete(TimeoutException.class);
+    }
+
+    @AfterTemplate
+    Mono<T> after(Mono<T> mono, Duration duration) {
+      return mono.timeout(duration, Mono.empty());
+    }
+  }
+
+  /**
+   * Prefer {@link Mono#timeout(Publisher, Mono)} over more contrived or less performant
+   * alternatives.
+   */
+  static final class MonoTimeoutPublisherMonoEmpty<T, S> {
+    @BeforeTemplate
+    Mono<T> before(Mono<T> mono, Publisher<S> other) {
+      return mono.timeout(other).onErrorComplete(TimeoutException.class);
+    }
+
+    @AfterTemplate
+    Mono<T> after(Mono<T> mono, Publisher<S> other) {
+      return mono.timeout(other, Mono.empty());
+    }
+  }
+
+  /**
+   * Prefer {@link Mono#timeout(Duration, Mono)} over more contrived or less performant
+   * alternatives.
+   */
+  static final class MonoTimeoutMonoJust<T> {
+    @BeforeTemplate
+    Mono<T> before(Mono<T> mono, Duration duration, T fallbackValue) {
+      return mono.timeout(duration).onErrorReturn(TimeoutException.class, fallbackValue);
+    }
+
+    @AfterTemplate
+    Mono<T> after(Mono<T> mono, Duration duration, T fallbackValue) {
+      return mono.timeout(duration, Mono.just(fallbackValue));
+    }
+  }
+
+  /**
+   * Prefer {@link Mono#timeout(Duration, Mono)} over more contrived or less performant
+   * alternatives.
+   */
+  static final class MonoTimeoutMonoFallback<T> {
+    @BeforeTemplate
+    Mono<T> before(Mono<T> mono, Duration duration, Mono<T> fallback) {
+      return mono.timeout(duration).onErrorResume(TimeoutException.class, e -> fallback);
```

Comment:
Let's validate that Refaster doesn't rewrite this case if `e` is used by `fallback`.